### PR TITLE
intel-oneapi-mpi: fix setup_dependent_build_environment

### PR DESCRIPTION
**Pull request by skosukhin**
The MPI compiler wrappers prefer `I_MPI_*` over `MPICH_*`, which means that if the former are set in the environment, the MPI compiler wrappers do not use the Spack compiler wrapper. This PR should fix that.

### CODE REVIEW

**Comment by skosukhin** [comment]
@climbfuji could you, please, check if this makes the workaround that you mentioned in https://github.com/spack/spack/pull/34580#issuecomment-1382326458 unnecessary? Thank you.

**Review by alalazo — APPROVED** [review]
Waiting for @rscohn2 and @climbfuji confirmation before merging

**Review by rscohn2 — APPROVED** [review]
Thanks

**Comment by climbfuji** [comment]
@skosukhin Is it sufficient to only set `I_MPI_*` or should we also set `MPICH_*` just because some weird package may be looking for this explicitly?

**Comment by skosukhin** [comment]
In my opinion, we set the variables for the MPI compiler wrappers of `intel-oneapi-mpi` to use the Spack compiler wrapper. No more, no less. Other packages are not supposed to look into these variables. And if they do, it's their problem, which they should work around.

**Comment by climbfuji** [comment]
> In my opinion, we set the variables for the MPI compiler wrappers of `intel-oneapi-mpi` to use the Spack compiler wrapper. No more, no less. Other packages are not supposed to look into these variables. And if they do, it's their problem, which they should work around.

Agreed, make sense.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**
